src/Compilers/Core/MSBuildTask/Vbc.cs

```diff
@@ -960,6 +954,37 @@ private bool InitializeHostCompiler(IVbcHostObject vbcHostObject)
             return true;
         }
 
+        // VbcHostObject doesn't support VB versions beyond 15,
+        // so the LangVersion will be passed through ICompilerOptionsHostObject.SetCompilerOptions instead
+        private static bool VbcHostObjectSupports(string langVersion)
+        {
+            if (langVersion == null)
+            {
+                return true;
+            }
+
+            var supportedList = new[]
+            {
+                "9", "9.0",
+                "10", "10.0",
+                "11", "11.0",
+                "12", "12.0",
+                "14", "14.0",
+                "15", "15.0"
+            };
+
+            int length = supportedList.Length;
+            for (int i = 0; i < length; i++)
+            {
+                if (supportedList[i].Equals(langVersion, StringComparison.OrdinalIgnoreCase))
```

Comment:
That was my initial inclination as well, but I didn't want to create more dependencies.